### PR TITLE
operator: Add memberlist IPv6 support

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Exposes the memberlist capability to run on IPv6 and dual-stack Kubernetes clusters.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: operator/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Main
 
+- [10562](https://github.com/grafana/loki/pull/10562) **periklis**:  Add memberlist IPv6 support
```
Comment (JoaoBraveCoding) [inline operator/CHANGELOG.md:3]: nit
```suggestion
- [10562](https://github.com/grafana/loki/pull/10562) **periklis**: Add memberlist IPv6 support
```

File: operator/internal/manifests/internal/config/loki-config.yaml [hunk 2]
```diff
@@ -187,9 +190,11 @@ limits_config:
 {{- with .GossipRing }}
 memberlist:
   abort_if_cluster_join_fails: true
+  {{- if not .EnableIPv6}}
```
Comment (JoaoBraveCoding) [inline operator/internal/manifests/internal/config/loki-config.yaml:193]: nit
```suggestion
  {{- if not .EnableIPv6 }}
```

File: operator/internal/manifests/internal/config/options.go [hunk 3]
```diff
@@ -60,6 +60,8 @@ type Address struct {
 
 // GossipRing defines the memberlist configuration
 type GossipRing struct {
+	// EnableIPv6 is optional memberlist IPv6 support
```
Comment (JoaoBraveCoding) [inline operator/internal/manifests/internal/config/options.go:63]: nit
```suggestion
	// EnableIPv6 is optional, memberlist IPv6 support
```

**Review by JoaoBraveCoding — COMMENTED** [review]
just some minor nits, lgtm

**Review by xperimental — COMMENTED** [review]
Works fine on an IPv6-only cluster and on a dual-stack IPv4/v6 cluster. Just some smaller comments.

File: operator/apis/loki/v1/v1.go [hunk 4]
```diff
@@ -81,6 +81,8 @@ var (
 	ErrReplicationFactorToZonesRatio = errors.New("replication factor is greater than the number of available zones")
 	// ErrReplicationSpecConflict when both the ReplicationSpec and depricated ReplicationFactor are used.
 	ErrReplicationSpecConflict = errors.New("replicationSpec and replicationFactor (deprecated) cannot be used at the same time")
+	// ErrIPv6InstanceAddrTypeNotAllowed when the default InstanceAddrType is used with enableIPv6.
+	ErrIPv6InstanceAddrTypeNotAllowed = errors.New("instanceAddrType `default` cannot be used with enableIPv6 at the same time")
```
Comment (xperimental) [inline operator/apis/loki/v1/v1.go:85]: Don't like the backticks in the output, but I don't think we have quoted any other values so far:

```suggestion
	ErrIPv6InstanceAddrTypeNotAllowed = errors.New(`instanceAddrType "default" cannot be used with enableIPv6 at the same time`)
```

**Review by xperimental — APPROVED** [review]

**Pull request merged**